cmake/multi_image.cmake

```diff
@@ -239,15 +239,14 @@ function(add_child_image_from_source)
       )
   endforeach()
 
+  set(preload_file ${CMAKE_BINARY_DIR}/${ACI_NAME}/child_image_preload.cmake)
+  file(WRITE ${preload_file} "# Generated file used for preloading a child image\n")
+
   unset(image_cmake_args)
   list(REMOVE_DUPLICATES SHARED_MULTI_IMAGE_VARIABLES)
   foreach(shared_var ${SHARED_MULTI_IMAGE_VARIABLES})
     if(DEFINED ${shared_var})
-      # Any  shared var that is a list must be escaped to ensure correct behaviour.
-      string(REPLACE \; \\\\\; val "${${shared_var}}")
-      list(APPEND image_cmake_args
-        -D${shared_var}=${val}
-        )
+        file(APPEND ${preload_file} "set(${shared_var} ${${shared_var}} CACHE INTERNAL \"\")\n")
```

Comment:
Should provide a good description here ?

Could be very helpful in when examine the CMakeCache and/or having to readback the cache inside CMake itself.
```suggestion
        file(APPEND ${preload_file} "set(${shared_var} ${${shared_var}} CACHE INTERNAL \"NCS child image controlled\")\n")
```